### PR TITLE
[chore] [exporter/elasticsearch] remove defunct config

### PR DESCRIPTION
**Pull request by axw**
**Description:**

Remove exporter/elasticsearch config fields & mentions in the README. Nothing has ever used this config.
They were added in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2324, but were not used.

If one _were_ to add fields to all documents, then I think that should be done with a processor.

**Link to tracking Issue:**

N/A

**Testing:**

N/A

**Documentation:**

N/A

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

**Review by carsonip — APPROVED** [review]

**Review by crobert-1 — COMMENTED** [review]
I'm slightly concerned that some users are using these config options and will run into errors on upgrade. I realize they don't do anything, but it may be good to add a breaking changelog just to let users know what's going on here.

I'll defer to code owners though.

**Comment by axw** [comment]
@crobert-1 thanks, good catch. I forgot that the config validation is strict, and would error on unknown keys. I'll add a breaking change notice.

**Review by carsonip — APPROVED** [review]

**Pull request merged**
